src/compiler/scala/tools/nsc/backend/jvm/BackendInterface.scala

```diff
@@ -601,6 +601,7 @@ abstract class BackendInterface extends BackendInterfaceDefinitions {
     def isTypeName: Boolean
     def isTermName: Boolean
     def startsWith(s: String): Boolean
+    def mangledString: String
```

Comment:
this change would break ScalaBackendInterface, as this method is not implemented there.
It would be good to add a CI here, that would track that we can still compile scalac out of this source.